extractor/filesystem/os/opkg/opkg.go

```diff
@@ -0,0 +1,285 @@
+		MaxFileSizeBytes:    e.maxFileSizeBytes,
+		IncludeNotInstalled: e.includeNotInstalled,
+	}
+}
+
+// Name of the extractor.
+func (e Extractor) Name() string { return Name }
+
+// Version of the extractor.
+func (e Extractor) Version() int { return 0 }
+
+// Requirements of the extractor.
+func (e Extractor) Requirements() *plugin.Capabilities { return &plugin.Capabilities{} }
+
+// FileRequired checks if the specified file matches OPKG status file pattern.
+func (e Extractor) FileRequired(api filesystem.FileAPI) bool {
+	path := api.Path()
+	if !fileRequired(path) {
+		return false
+	}
+
+	fileinfo, err := api.Stat()
+	if err != nil {
+		return false
+	}
+	if e.maxFileSizeBytes > 0 && fileinfo.Size() > e.maxFileSizeBytes {
+		e.reportFileRequired(path, fileinfo.Size(), stats.FileRequiredResultSizeLimitExceeded)
+		return false
+	}
+
+	e.reportFileRequired(path, fileinfo.Size(), stats.FileRequiredResultOK)
+	return true
+}
+
+func fileRequired(path string) bool {
+	// status file
+	return filepath.ToSlash(path) == "usr/lib/opkg/status"
+}
+
+func (e Extractor) reportFileRequired(path string, fileSizeBytes int64, result stats.FileRequiredResult) {
+	if e.stats == nil {
+		return
+	}
+	e.stats.AfterFileRequired(e.Name(), &stats.FileRequiredStats{
+		Path:          path,
+		Result:        result,
+		FileSizeBytes: fileSizeBytes,
+	})
+}
+
+// Extract extracts packages from OPKG status files passed through the scan input.
+func (e Extractor) Extract(ctx context.Context, input *filesystem.ScanInput) ([]*extractor.Inventory, error) {
+	inventory, err := e.extractFromInput(ctx, input)
+	if e.stats != nil {
+		var fileSizeBytes int64
+		if input.Info != nil {
+			fileSizeBytes = input.Info.Size()
+		}
+		e.stats.AfterFileExtracted(e.Name(), &stats.FileExtractedStats{
+			Path:          input.Path,
+			Result:        filesystem.ExtractorErrorToFileExtractedResult(err),
+			FileSizeBytes: fileSizeBytes,
+		})
+	}
+	return inventory, err
+}
+
+func (e Extractor) extractFromInput(ctx context.Context, input *filesystem.ScanInput) ([]*extractor.Inventory, error) {
+	m, err := osrelease.GetOSRelease(input.FS)
+	if err != nil {
+		log.Errorf("osrelease.ParseOsRelease(): %v", err)
+	}
+
+	rd := textproto.NewReader(bufio.NewReader(input.Reader))
+	pkgs := []*extractor.Inventory{}
+	for eof := false; !eof; {
+		// Return if canceled or exceeding deadline.
+		if err := ctx.Err(); err != nil {
+			return pkgs, fmt.Errorf("%s halted at %q because of context error: %v", e.Name(), input.Path, err)
+		}
+
+		h, err := rd.ReadMIMEHeader()
+		if err != nil {
+			if errors.Is(err, io.EOF) {
+				// We might still have one more line of data
+				// so return only after it's been parsed.
+				eof = true
+			} else {
+				return pkgs, err
+			}
+
+		}
+
+		// Skip empty lines
+		if len(h) == 0 {
+			continue
+		}
+
+		// Skip packages based on status
+		status := h.Get("Status")
+		if !e.includeNotInstalled && status != "install ok installed" {
```

Comment:
if the package is `deinstall ok installed` it'll still be installed, just marked for deinstallation. We should fine those packages as well.
I think we can just reuse the dpkg extractor's statusInstalled() function for this.